x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/chunking/EmbeddingRequestChunker.java

```diff
@@ -82,11 +85,35 @@ public EmbeddingRequestChunker(
         this.wordsPerChunk = wordsPerChunk;
         this.chunkOverlap = chunkOverlap;
         this.embeddingType = embeddingType;
+        this.chunkingSettings = null;
+        splitIntoBatchedRequests(inputs);
+    }
+
+    public EmbeddingRequestChunker(
+        List<String> inputs,
+        int maxNumberOfInputsPerBatch,
+        EmbeddingType embeddingType,
+        ChunkingSettings chunkingSettings
+    ) {
+        this.maxNumberOfInputsPerBatch = maxNumberOfInputsPerBatch;
+        this.wordsPerChunk = DEFAULT_WORDS_PER_CHUNK; // Can be removed after ChunkingConfigurationFeatureFlag is enabled
+        this.chunkOverlap = DEFAULT_CHUNK_OVERLAP; // Can be removed after ChunkingConfigurationFeatureFlag is enabled
+        this.embeddingType = embeddingType;
+        this.chunkingSettings = chunkingSettings;
         splitIntoBatchedRequests(inputs);
     }
 
     private void splitIntoBatchedRequests(List<String> inputs) {
-        var chunker = new WordBoundaryChunker();
+        Function<String, List<String>> chunkFunction;
+        if (chunkingSettings != null) {
+            var chunkingStrategy = chunkingSettings.getChunkingStrategy() != null ? chunkingSettings.getChunkingStrategy() : null;
+            var chunker = ChunkerBuilder.fromChunkingStrategy(chunkingStrategy);
```

Comment:
`chunkingSettings.getChunkingStrategy() != null ? chunkingSettings.getChunkingStrategy() : null;` 

if `chunkingSettings.getChunkingStrategy() == null` then `chunkingStrategy` is set to `null`, this can be simplified since `ChunkerBuilder.fromChunkingStrategy` handles the null case

```suggestion
            var chunker = ChunkerBuilder.fromChunkingStrategy(chunkingSettings.getChunkingStrategy());
```